components/console-backend-service/internal/domain/eventing/trigger/service_test.go

```diff
@@ -0,0 +1,421 @@
+			errMatcher:     gomega.HaveOccurred(),
+			triggerMatcher: gomega.BeNil(),
+		},
+	} {
+		t.Run(testName, func(t *testing.T) {
+			//given
+			g := gomega.NewGomegaWithT(t)
+
+			//when
+			service := NewService(serviceFactory)
+			testingUtils.WaitForInformerStartAtMost(t, time.Second, service.GetInformer())
+
+			created, err := service.Create(testData.trigger)
+
+			//then
+			g.Expect(err).To(testData.errMatcher)
+			g.Expect(created).To(testData.triggerMatcher)
+		})
+	}
+}
+
+func TestTriggerService_CreateMany(t *testing.T) {
+	serviceFactory, err := resourceFake.NewFakeServiceFactory(v1alpha1.AddToScheme)
+	require.NoError(t, err)
+
+	url := "www.test.com"
+
+	for testName, testData := range map[string]struct {
+		triggers       []*v1alpha1.Trigger
+		errMatcher     types.GomegaMatcher
+		triggerMatcher types.GomegaMatcher
+	}{
+		"Success": {
+			triggers: []*v1alpha1.Trigger{
+				fixTriggerWithUri("a1", "a", url),
+				fixTriggerWithUri("a2", "a", url),
+				fixTriggerWithRef("a1", "b", "refA1", "refA"),
+			},
+			errMatcher:     gomega.BeNil(),
+			triggerMatcher: gomega.HaveLen(3),
+		},
+		"Empty": {
+			triggers:       []*v1alpha1.Trigger{},
+			errMatcher:     gomega.BeNil(),
+			triggerMatcher: gomega.HaveLen(0),
+		},
+		"Nil": {
+			triggers:       nil,
+			errMatcher:     gomega.BeNil(),
+			triggerMatcher: gomega.HaveLen(0),
+		},
+	} {
+		t.Run(testName, func(t *testing.T) {
+			//given
+			g := gomega.NewGomegaWithT(t)
+
+			//when
+			service := NewService(serviceFactory)
+			testingUtils.WaitForInformerStartAtMost(t, time.Second, service.GetInformer())
+
+			created, err := service.CreateMany(testData.triggers)
+
+			//then
+			g.Expect(err).To(testData.errMatcher)
+			g.Expect(created).To(testData.triggerMatcher)
+		})
+	}
+}
+
+func TestTriggerService_Delete(t *testing.T) {
+	url := "www.test.com"
+	serviceFactory, err := resourceFake.NewFakeServiceFactory(v1alpha1.AddToScheme,
+		fixTriggerWithUri("a1", "a", url),
+		fixTriggerWithUri("a2", "a", url),
+		fixTriggerWithRef("a1", "b", "refA1", "refA"),
+	)
+	require.NoError(t, err)
+
+	for testName, testData := range map[string]struct {
+		trigger    gqlschema.TriggerMetadataInput
+		errMatcher types.GomegaMatcher
+	}{
+		"Success": {
+			trigger:    gqlschema.TriggerMetadataInput{Name: "a2", Namespace: "a"},
+			errMatcher: gomega.BeNil(),
+		},
+		"Without namespace": {
+			trigger:    gqlschema.TriggerMetadataInput{Name: "a2"},
+			errMatcher: gomega.HaveOccurred(),
+		},
+		"Without name": {
+			trigger:    gqlschema.TriggerMetadataInput{Namespace: "a"},
+			errMatcher: gomega.HaveOccurred(),
+		},
+		"empty": {
+			trigger:    gqlschema.TriggerMetadataInput{},
+			errMatcher: gomega.HaveOccurred(),
+		},
+	} {
+		t.Run(testName, func(t *testing.T) {
+			//given
+			g := gomega.NewGomegaWithT(t)
+
+			//when
+			service := NewService(serviceFactory)
+			testingUtils.WaitForInformerStartAtMost(t, time.Second, service.GetInformer())
```

Comment:
Please avoid using `service.GetInformer()` something like here https://github.com/kyma-project/kyma/blob/master/components/console-backend-service/internal/domain/rafter/asset_service_test.go#L166

Reply:
Here and also in other tests